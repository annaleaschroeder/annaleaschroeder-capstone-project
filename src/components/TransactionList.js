import React from 'react'
import Transaction from './Transaction'

export default function TransactionList({ transactions }) {
  return (
    <section>
      {transactions.map(({ timestamp, number, id }) => (
        <Transaction key={id} createdAt={timestamp} newTransaction={number} />
      ))}
    </section>
  )
}
