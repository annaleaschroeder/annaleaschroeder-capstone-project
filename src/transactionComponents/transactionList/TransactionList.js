import React from 'react'
import Transaction from '../singleTransaction/Transaction'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

TransactionList.propTypes = {
  transactions: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default function TransactionList({ deleteTransaction, transactions }) {
  return (
    <StyledList data-cy="transactionsList">
      {transactions.map(({ timestamp, value, id, type, notes, tag }) => (
        <li key={id}>
          <Transaction
            deleteTransaction={(event) => {
              event.stopPropagation()
              if (window.confirm('Are you sure?')) {
                deleteTransaction(id)
              }
            }}
            createdAt={timestamp}
            newTransaction={value}
            type={type}
            notes={notes}
            tag={tag}
            id={id}
          />
        </li>
      ))}
    </StyledList>
  )
}

const StyledList = styled.ul`
  list-style: none;
  padding-left: 0;
`
