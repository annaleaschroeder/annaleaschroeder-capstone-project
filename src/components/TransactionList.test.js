import React from 'react'
import TransactionList from './TransactionList'
import renderer from 'react-test-renderer'

describe('Lists all incoming Transactions', () => {
  it('renders correctly', () => {
    const transaction = [
      { key: '1', createdAt: '9.1.2020', newTransaction: '15,99' },
    ]
    const key = '1'
    const tree = renderer.create(
      <TransactionList key={key} transactions={transaction} />
    )
    expect(tree).toMatchSnapshot()
  })
})
