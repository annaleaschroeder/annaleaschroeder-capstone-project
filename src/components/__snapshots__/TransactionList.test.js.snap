// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists all incoming Transactions renders correctly 1`] = `
<section>
  <section>
    <p>
      undefined undefined
    </p>
  </section>
</section>
`;
